### PR TITLE
Xiao F add floating and dismissable team notification header bar

### PR DESCRIPTION
**Pull request by tsunami776**
# Description
CleanShot 2023-11-22 at 16 28 21

## Related PRS (if any):
N/A

## Main changes explained:
- Update file `src/components/Dashboard/Dashboard.jsx` for including new header bar and its time logic
- Update various other files to implement styling of the header bar

## How to test:
1. Check into current branch
2. Do `npm install` and `npm run start:local` to run this PR locally
3. Log in as volunteer/manager on a team (Check if a person is on a team by going to its profile)
4. Go to dashboard -> Should see a blue header bar with different text according to the role
5. Dismiss the header and the header will not show up again until next Thursday
6. Log in as any other roles -> Should not see the blue header
7. Change the system time to next Friday
8. Log out and log in again as volunteer/manager ->Should see the blue header again

## Screenshots or videos of changes:

For volunteer on a team:
CleanShot 2023-11-22 at 17 38 31

For manager on a team:
CleanShot 2023-11-22 at 17 38 59

Test video:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/43768723/ddb4201c-0162-4fc5-98c0-e9e69399a00f

## Note:
N/A

### CODE REVIEW

**Review by scoopstroop92 — DISMISSED** [review]
Hello @tsunami776  good job on the functionality 

Since I tested the functionality today (Friday/ Saturday ) I didnt have to update the system timings. So the notification works as expected

**Volunteer**
Vol-1575-Friday
**Manager**
Vol-1575-lasttuesady

However, When i tried changing the system date to a previous/ future date  thats not a thursday/friday/saturday , I still see the notification. Is that expected ?
Vol-1575-lasttuesady
image

**Comment by ptrpengdev** [comment]
Hi @tsunami776, I tested with the volunteer role. The message was displayed correctly and dismissed the header and the header did not show up upon re-login. I also tested with the Admin role. No message is shown.

Overall, the feature worked as expected. Good job! Please **remember to pull the latest code** and **resolve any conflict** before merging!

Initial Login with the Volunteer role.
image

Dismissed message and re-login with Volunteer role. 
image

Admin role:
image

**Comment by tsunami776** [comment]
Hi @ramyaram2092, thanks for reviewing my pull request. Yes, it is expected to see the notification again if you do not dismiss the notification. If you dismiss the notification, it will not show up again until next Thursday.

**Comment by tsunami776** [comment]
Hi @ptrpengdev, thank you for reviewing my pull request and reminding me to resolve the conflicts. I resolved the conflicts and merged dev branch into my branch. Please pull again and check the functionalities still work.

**Review by sucheta90 — DISMISSED** [review]
Hi @tsunami776 , tested three different roles - Manager, Assistant Manager and Volunteer. The code works as expected. 

- As **MANAGER**

PR#1575_review_tested_as_manager

- As **Assistant Manager**
PR#1575_review_tested_assistantManager

- As **Volunteer**
PR#1575_review_tested_as_volunteer

- System's time stamp were changed for each use case. Message goes away when dismissed and doesn't come back until next Thursday. To test, checked week **11/16/2023** and **11/23/2023** as can be seen on the screenshots.

**Review by Alforoan — DISMISSED** [review]
The changes seem to work for volunteers. Nice!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/118868270/9c7713c3-e875-4427-ba63-3d6ec0621977

**Comment by DragonSlayer2401** [comment]
I tested the code and it appears to be working correctly. The message only popped up on my manager and volunteer account as intended. After closing the message it didn't come back when I refreshed as intended. I did have to set my system time to December 22nd but that may of been because I already tested it on December 8th and closed the message. Please see the overview video below for more information:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/102876103/5c7a3d72-2859-4c1f-b2e6-38aa12ebef94

**Comment by tsunami776** [comment]
Hi @DragonSlayer2401, thanks for your review and pointing out an edge case! I pushed a new commit fixed it by checking if user are going back in time.

**Review by ZichanYang324 — DISMISSED** [review]
Hi @tsunami776, I just logged in as the manager and tested the feature for the blue head bar. After I dismissed it, it disappeared as expected. When I logged out and changed the local PC time to next Friday, the blue head bar showed up again, the same as the requirement.

All the features match to the goal and there is no problem, well done!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/86720318/960c2fb5-68f0-4c78-8f35-28fdb6b4254e

File: src/App.css [hunk 1]
```diff
@@ -115,9 +114,31 @@ nav ul.navbar-nav li:last-child>div {
...
 }
 
 tr {
   content-visibility: auto;
-}
\ No newline at end of file
+}
+
+.floating-header {
+  position: fixed; /* Keeps the button in the same position on the screen */
```
Comment (GKD890) [inline src/App.css:125]: Hi, I was a bit curious about the responsive effect, thus have a try. I don't think using `fixed` is good here as you may check how it looks like in the video. 
The first half is the `development` branch while when I move my mouse around the **element** in the right panel, it's this PR and the header is almost like a nav bar. I guess it's not part of this PR? If I'm wrong, please correct me.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/57503876/1dcd6b2e-82bb-489a-bc41-7461f70f7752

**Review by keyunhuangg — DISMISSED** [review]
1) Log in as volunteer and manager on a team and see a blue header bar with the right role. Then dismisses the header, the header did not show up after re-logging back in.
2) Changed the system time to 12/29. Logged in as volunteer and manager and both blue header show up again.
Screenshot 2023-12-21 at 3 16 10 PM
Screenshot 2023-12-21 at 3 15 09 PM

**Comment by mianmiantea2019** [comment]
Hi @tsunami776,

I've tested the feature with the volunteer role and manager role, and everything seems to be working as expected. The message was displayed correctly, and after dismissing the header, it didn't reappear upon re-login, which is the intended behavior.

Additionally, I manually adjusted the system time to next week, and as expected, the message popped up when logging in as a volunteer or manager. Great job! 

Login with the Volunteer role.
image

Login with Manager role:
image

image

Dismissed message and re-login with Manager role.
image

**Review by ljrirene — DISMISSED** [review]
Hi Xiaofei, the PR works as expected. 
When I logged in as a manager user:

Uploading Screen Recording 2023-12-23 at 19.02.46.mov…



When I logged in as a volunteer user:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/34266880/5d3f5b51-8c85-407e-9f68-4b5dc481d9a9

**Comment by yuhuimin1996** [comment]
Hi Xiao,
I've tested this PR with both volunteer account and also manager account. Both of these two kinds of users worked as you described. When they are in a team, the blue header will appear on the dashboard. After click 'x' button, the blue header hint will not show up again after refreshing the page or log in again. After I set the time to next Friday, the blue header hint will appear again on the dashboard. Good job! Please see more details in the videos:
volunteer:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108281570/5cc57012-de06-48e6-8a83-e15791881aa4

manager:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108281570/e87b3f40-0ced-4a82-b3c2-69a3d516a4e0

**Review by pshereen — CHANGES_REQUESTED** [review]
Tested the PR using admin role and volunteer role.
It is working as per the requirement. 

However, if I change the system time to a date past next Thursday, the dismissed notification is not coming back. For e.g. I gave a date after one month and the notification was not showing up. 

Screenshot 2024-01-18 183126


Screenshot 2024-02-17 183039

**Review by rusalshrestha98 — APPROVED** [review]
I was able to verify that the banner appeared with the correct text respective of the role (manager or volunteer) for each new week, and that it would disappear for the rest of the week once the user clicked the "X" button to close it. Great job!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/48131438/89c1fba7-eaf6-46b9-a986-868dd84b6b4b

**Review by X1aoZhang99 — APPROVED** [review]
Hi Xiao,

I've tested this PR with both volunteer users and manager users within the same team. Both user types performed as you described. 

However, I did notice a minor issue. When I logged in as a manager for the first time, the blue banner at the top didn't appear until I refreshed the page. This behavior only occurred after I had initially logged in as a volunteer and then as a manager.

Please take a look at the video for reference.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/113070229/e3fa32e6-6566-4818-b807-43c03a8ea262

**Review by Renanluizssx — DISMISSED** [review]
Hi @tsunami776, if i understood, when i entered in first time, it will show me the blue box message, and if i dismiss it, it didn't show me again, but if i change the system time to next friday it will show me right? i changed my time to next friday, but the blue message box didn't appear to me, see my review below:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/91035340/f66bd6b2-7628-4885-a00b-36bc37eb4bcb

**Review by Sophie-lei — APPROVED** [review]
Hey, I test this PR with both of the volunteer roles, the manager, and the owner role. It looks like it was expected. The blue notification is shown for the volunteer and manager role after I changed the current date to Friday. The blue notification is not shown for the owner role. 
Screenshot 2024-02-16 at 13 35 28

Screenshot 2024-01-18 at 12 55 14
Screenshot 2024-01-18 at 12 55 41

**Review by 666saofeng — APPROVED** [review]
I tested as manager and volunteer role. It works as expected.
test1
test2
test3
test4

**Comment by nathanah** [comment]
> Hi @tsunami776, if i understood, when i entered in first time, it will show me the blue box message, and if i dismiss it, it didn't show me again, but if i change the system time to next friday it will show me right? i changed my time to next friday, but the blue message box didn't appear to me, see my review below:
> 
>  2024-01-27.00-55-32.mov

Your volunteer and manager accounts are not on a team and should not be getting the notification. You're also using a mentor account instead of a manager account in the video, which shouldn't receive the notification either.

**Comment by xmyoot** [comment]
I reviewed the following PR as a Manager Role part of "Hector Team" and I didn't notice the blue notification when changing the date to Friday please reference: 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/15240538/b59af102-a298-4494-ae28-02674d651c22

I then changed roles to Volunteer and again ensure the User was on the team, and changed the date to Friday again but didn't observe the blue notification: 

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/15240538/969d5263-4ebd-49aa-8ec9-25b81e71613e

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**